docs/search-engine-architecture-details.md

```diff
@@ -0,0 +1,28 @@
+# Search Engine Architecture
+
+## Asynch Processes
+
+1. Indexations of new, updated and deleted documents;
+1. Search documents.
+
+
+## Abstarct Search Service Module
+
+Provides ability to add custom types of documents:
+
+1. ClassIndexDocumentConfiguration;
+1. Interface Index Document Changes Provider- to detect modified documents;
+1. Interface ISerach Request Builder- to transform Search Criteria to Search Request.
```

Comment:
Interface `ISearchRequestBuilder` - to

Proper name + ISe**ar**ch

Recommendation: you could use back apostrophe to surround code in Github markdown, like `SomeClassName`